e2e/wdio/platform/tests/date-time-picker.e2e-spec.ts

```diff
@@ -0,0 +1,200 @@
+        dateTimePickerPage.open();
+    }, 1);
+
+    afterEach(() => {
+        refreshPage();
+        waitForPresent(inputDatePicker);
+    }, 1);
+
+    it('Verify in all the form factor user is able to see the date picker button and input field ', () => {
+        const buttons = elementArray(buttonDatePicker);
+        const inputs = elementArray(inputDatePicker);
+        expect(buttons.length).toEqual(inputs.length);
+        for (let i = 1; i < buttons.length; i++) {
+            waitForElDisplayed(buttonDatePicker);
+            waitForElDisplayed(inputDatePicker);
+        }
+    });
+
+    it('Verify on click on the date picker button', () => {
+        const activeButtons = elementArray(activeButtonDateTimePicker);
+        for (let i = 1; i < activeButtons.length; i++) {
+            sendKeys(['Escape']);
+            scrollIntoView(activeButtonDateTimePicker, i);
+            click(activeButtonDateTimePicker, i);
+            waitForElDisplayed(calendarExpanded);
+        }
+    });
+
+    it('Verify From the day view on the calendar, clicking or tapping a year', () => {
+        sendKeys(['Escape']);
+        scrollIntoView(bottomPage);
+        click(activeButtonDateTimePicker, 1);
+        expect(waitForElDisplayed(calendarExpanded)).toBe(true);
+        scrollIntoView(topPage);
+        scrollIntoView(buttonSelectYear);
+        click(buttonSelectYear);
+        expect(waitForElDisplayed(calendarYearsSection)).toBe(true);
+        click(dateTimePickerPage.yearInCalendarByValue(dateTimePickerTestData.year2030));
+        expect(getText(currentYear)).toBe(dateTimePickerTestData.year2030.toString());
+    });
+
+    it('Verify by default today date is focused', () => {
+        const activeButtons = elementArray(activeButtonDateTimePicker);
+        for (let i = 0; i < activeButtons.length; i++) {
+            if (i !== 2 && i !== 6) {  // other default days in these calendars
+                sendKeys(['Escape']);
+                scrollIntoView(activeButtonDateTimePicker, i);
+                click(activeButtonDateTimePicker, i);
+                waitForElDisplayed(calendarExpanded);
+                expect(getText(currentDay, 0)).toBe(new Date().getDate().toString());
+            }
+        }
+    });
+
+    it('Verify on click on the input field ', () => {
+        const activeInputs = elementArray(activeInputDateTimePicker);
+        for (let i = 0; i < activeInputs.length; i++) {
+            sendKeys(['Escape']);
+            scrollIntoView(activeInputDateTimePicker, i);
+            setValue(activeInputDateTimePicker, datePickerTestData.text, i);
+            expect(getValue(activeInputDateTimePicker, i)).toBe(datePickerTestData.text);
+        }
+    });
+
+    it('Verify date input field have placeholder', () => {
+        const inputs = elementArray(inputDatePicker);
+        for (let i = 0; i < inputs.length; i++) {
+            expect(getAttributeByName(inputDatePicker, 'placeholder', i)).toBeDefined();
```

Comment:
toBeDefined only checks the returned response isn't 'undefined', if the returned value is null or empty string it will pass. Depending on what you are checking, might need to change this one.